editor/editor_help_search.cpp

```diff
@@ -331,88 +345,141 @@ bool EditorHelpSearch::Runner::_phase_match_classes_init() {
-						match.properties.push_back(const_cast<DocData::PropertyDoc *>(&class_doc.properties[i]));
+				for (int i = 0; i < class_doc->properties.size(); i++) {
+					if (_all_terms_in_name(class_doc->properties[i].name)) {
+						match.properties.push_back(const_cast<DocData::PropertyDoc *>(&class_doc->properties[i]));
 					}
 				}
 			}
 			if (search_flags & SEARCH_THEME_ITEMS) {
-				for (int i = 0; i < class_doc.theme_properties.size(); i++) {
-					if (_all_terms_in_name(class_doc.theme_properties[i].name)) {
-						match.theme_properties.push_back(const_cast<DocData::ThemeItemDoc *>(&class_doc.theme_properties[i]));
+				for (int i = 0; i < class_doc->theme_properties.size(); i++) {
+					if (_all_terms_in_name(class_doc->theme_properties[i].name)) {
+						match.theme_properties.push_back(const_cast<DocData::ThemeItemDoc *>(&class_doc->theme_properties[i]));
 					}
 				}
 			}
 			if (search_flags & SEARCH_ANNOTATIONS) {
-				for (int i = 0; i < class_doc.annotations.size(); i++) {
-					if (_match_string(term, class_doc.annotations[i].name)) {
-						match.annotations.push_back(const_cast<DocData::MethodDoc *>(&class_doc.annotations[i]));
+				for (int i = 0; i < class_doc->annotations.size(); i++) {
+					if (_match_string(term, class_doc->annotations[i].name)) {
+						match.annotations.push_back(const_cast<DocData::MethodDoc *>(&class_doc->annotations[i]));
 					}
 				}
 			}
 		}
-		matches[class_doc.name] = match;
+		matches[class_doc->name] = match;
 	}
 
-	++iterator_doc;
-	return !iterator_doc;
+	if (iterator_doc) {
+		++iterator_doc;
+		return !iterator_doc;
+	} else {
+		if (!iterator_stack.is_empty()) {
+			if (iterator_stack[iterator_stack.size() - 1]) {
+				iterator_stack.write[iterator_stack.size() - 1] = iterator_stack.write[iterator_stack.size() - 1]->next();
+			}
+			if (!iterator_stack[iterator_stack.size() - 1]) {
+				iterator_stack.resize(iterator_stack.size() - 1);
+			}
+		}
+
+		if (class_doc && EditorHelp::get_doc_data()->inheriting.has(class_doc->name)) {
+			iterator_stack.push_back(EditorHelp::get_doc_data()->inheriting[class_doc->name].front());
+		}
+
+		return iterator_stack.is_empty();
+	}
+}
+
+void EditorHelpSearch::Runner::_populate_cache() {
+	root_item = results_tree->get_root();
+
+	if (root_item) {
+		LocalVector<TreeItem *> stack;
+
+		// Add children of root item to stack.
+		for (TreeItem *child = root_item->get_first_child(); child; child = child->get_next()) {
+			stack.push_back(child);
+		}
+
+		// Traverse stack and cache items.
+		while (!stack.is_empty()) {
+			TreeItem *cur_item = stack[stack.size() - 1];
+			stack.resize(stack.size() - 1);
+
+			// Add to the cache.
+			tree_cache->item_cache.insert(cur_item->get_metadata(0).operator String(), cur_item);
+
+			// Add any children to the stack.
+			for (TreeItem *child = cur_item->get_first_child(); child; child = child->get_next()) {
+				stack.push_back(child);
+			}
+
+			// Remove from parent (if any).
```

Comment:
```suggestion
			// Remove from parent.
```
Leftover from earlier version, will fix